managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/ManagedCursorImpl.java

```diff
@@ -1705,7 +1710,8 @@ public void asyncClose(final AsyncCallbacks.CloseCallback callback, final Object
 
         lock.readLock().lock();
         try {
-            if (!individualDeletedMessages.isEmpty()) {
+            if (cursorLedger != null && protobufFormat == ZNodeProtobufFormat.Text
```

Comment:
Well, that's the same behavior as if the `ledger.addEntry()` fails, we fail the `cursor.close()` operation and then it will recover from the cursor ledger next time.

The reason to write directly on the z-node was to speed-up the cursor recovery by needing just one zk read to be completed.